source/deployment/deployContracts.ts

```diff
@@ -19,9 +18,13 @@ async function doWork() {
     await contractDeployer.deploy();
 }
 
-doWork().then(() => {
-    process.exit();
-}).catch(error => {
-    console.log(error);
-    process.exit();
-});
+if (require.main === module) {
```

Comment:
The original intent with files in the `deployment` folder was that they would never be included as a library, they were meant to be "dumb wrappers" around library code that did all of the work.  Wrapping in `if (require.main === module)` implies that this may be used as a library.

My recommendation would be that if you find the business logic in this file to be valuable, then extract it into a library and leave this as a "dumb wrapper".  In this case, it looks like the thing you want is the composition of the above 6 lines, perhaps we could just add a helper function to `ContractDeployer` that does this composition for you?

Alternatively, if you think the pattern of separating library code from script code is stupid, feel free to do whatever you want since I no longer own anything.  😉

Reply:
It may make sense to move that into ContractDeployer... I'll take a look.